firedrake/functionspacedata.py

```diff
@@ -552,7 +552,7 @@ def get_shared_data(mesh, fiat_element):
     """
     if not isinstance(mesh, mesh_mod.MeshTopology):
         raise ValueError("%s is not a MeshTopology" % mesh)
-    if not isinstance(fiat_element, FIAT.finite_element.FiniteElement):
+    if not isinstance(fiat_element, (FIAT.finite_element.CiarletElement, FIAT.finite_element.FiniteElement)):
```

Comment:
Doesn't `CiarletElement` inherit from `FiniteElement`? It should. If it does, it should sufficient to check against FiniteElement only.